flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/IcebergTableSink.java

```diff
@@ -31,7 +31,7 @@
 import org.apache.flink.table.connector.sink.abilities.SupportsPartitioning;
 import org.apache.flink.types.RowKind;
 import org.apache.flink.util.Preconditions;
-import org.apache.iceberg.flink.sink.FlinkSink;
+import org.apache.iceberg.flink.sink.v2.FlinkSink;
```

Comment:
I would recommend let's not switch to the new impl in the first release as we are still learning the behaviors of new sink interface. it is better to provide a config for users to the new sink impl

Reply:
Done, but I don't think `iceberg-sink` is a good name. Do you have a better suggestion?

```
  // enable iceberg new sink
  public static final ConfigOption<Boolean> ICEBERG_SINK =
      ConfigOptions.key("iceberg-sink")
          .booleanType().defaultValue(false);
```

Reply:
Maybe `table.exec.iceberg.use-flip143-sink`.

Here is the source config from my r[ecently opened PR](https://github.com/apache/iceberg/pull/5318/files#diff-dcbfb695653a4eb40200eef1889e2526122c3748c6846ef40d90987bdf0a7563). 
```
public static final ConfigOption<Boolean> TABLE_EXEC_ICEBERG_USE_FLIP27_SOURCE =
      ConfigOptions.key("table.exec.iceberg.use-flip27-source")
          .booleanType()
          .defaultValue(false)
          .withDescription("Use the FLIP-27 based Iceberg source implementation.");
```

Reply:
nice.